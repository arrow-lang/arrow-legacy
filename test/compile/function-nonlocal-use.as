let value;
def get() -> int { return value; }

value = 20; // ok
let box = get(); // ok
