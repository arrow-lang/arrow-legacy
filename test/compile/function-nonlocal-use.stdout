@value = internal global i128 0
@value1 = internal global i128 0

define internal void @function-nonlocal-use..init() {
  store i128 20, i128* @value
  %1 = call i128 @get()
  store i128 %1, i128* @value1
  %2 = load i128* @value1
  call void @exit(i128 %2)
  ret void
}

define internal i128 @get() {
  %1 = load i128* @value
  ret i128 %1
}

declare void @exit(i128 signext)

define i32 @main(i32, i8**, i8**) {
  call void @function-nonlocal-use..init()
  ret i32 0
}
